### PR TITLE
Fix/profile/solve editprofile bug

### PR DESCRIPTION
**Pull request by Harrish92**
# Solve editprofile bug

## Description
This PR introduces a solution to the crash caused by the photo gallery in `EditProfile` .  It closes issue #351 .

## Changes
- Verified that variables are not null.
- Made sure that the code is safely encapsulated.

#### Modified
- `app/src/main/java/com/android/periodpals/ui/components/ProfileComponents.kt`
- `app/src/main/java/com/android/periodpals/ui/profile/EditProfile.kt`
- `app/src/main/java/com/android/periodpals/ui/profile/ProfileScreen.kt`

## Testing
- This PR does not include testing, since it refactors pre-written code.

### CODE REVIEW

File: app/src/main/java/com/android/periodpals/ui/components/ProfileComponents.kt [hunk 1]
```diff
@@ -241,6 +241,11 @@ fun ProfileSaveButton(
           Toast.makeText(context, errorMessage, Toast.LENGTH_SHORT).show()
           return@Button
         }
+        if (preferredDistance == null) {
```
Comment (charliemangano) [inline app/src/main/java/com/android/periodpals/ui/components/ProfileComponents.kt:244]: Good reflex!

File: app/src/test/java/com/android/periodpals/ui/profile/EditProfileTest.kt [hunk 2]
```diff
@@ -118,6 +118,7 @@ class EditProfileTest {
 
     `when`(navigationActions.currentRoute()).thenReturn(Route.PROFILE)
     `when`(userViewModel.formState).thenReturn(formState)
+    `when`(userViewModel.avatar).thenReturn(mutableStateOf(ByteArray(0)))
```
Comment (charliemangano) [inline app/src/test/java/com/android/periodpals/ui/profile/EditProfileTest.kt:121]: Nice catch!

**Review by charliemangano — APPROVED** [review]
Great job introducing better practices in this code, it's always good to check null values and here it even fixes a bug!

**Review by coaguila — APPROVED** [review]
LGTM

**Pull request merged**